client/js/_player/components/common/recorder_timer.jsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { Line } from 'rc-progress';
+
+class RecorderTimer extends React.Component {
+  static propTypes = {
+    // Maximum audio recording length in seconds
+    timeout: React.PropTypes.number,
+  };
+
+  constructor() {
+    super();
+    this.state = {
+      minsCt       : 0,
+      secsCt       : 0,
+      secsStringCt : '00',
+      prcntCt      : 0
+    };
+  }
+
+  componentDidMount() {
+    this.handleTimerCount();
+  }
+
+  componentWillUnmount() {
+    this.clearTimerCount();
+  }
+
+  tick() {
+    this.setState(prevState => ({
+      prcntCt: Math.round(((prevState.prcntCt + 1) / this.props.timeout) * 100),
+    }));
+
+    if (this.state.secsCt < 9) {
+      this.setState(prevState => ({
+        secsCt       : prevState.secsCt + 1,
+        secsStringCt : `0${prevState.secsCt + 1}`,
+      }));
+    } else if (this.state.secsCt === 59) {
+      this.setState({
+        minsCt       : 1,
+        secsCt       : 0,
+        secsStringCt : '00',
+      });
+    } else {
+      this.setState(prevState => ({
+        secsCt       : prevState.secsCt + 1,
+        secsStringCt : prevState.secsCt + 1,
+      }));
+    }
+  }
+
+  handleTimerCount = () => {
```

Comment:
Is the way I've coded this not a best practice? I certainly welcome tips on JS/React best practices.